### DISCUSSION TITLE
Different number of columns per row

### DISCUSSION THREAD

**Discussion by mydnic** [filamentphp/filament#8774] [Help] [answered]
Hello,
In a custom page, is it possible to get different columns per row ? 

Exemple : I want to display 6 widgets on my page.
2 widgets that are 1/2 width on the first row
4 widgets that are 1/4 width on the second row

I tried this 
```php
class Dashboard extends \Filament\Pages\Dashboard
{
    public function getColumns(): int | string | array
    {
        return [2, 4];
    }
}
```

but it doesn't work, I think return an array is only for breakpoints.

I didn't find anything in the documentation.

Thanks in advance!

**Comment by VincvanAs** [comment, marked as answer]
You could just set the number on the dashboard page on the max columns you prefer, and then define the columnspan on every widget to fill in those spots 

protected int | string | array $columnSpan = 2;

**Reply by mydnic** [reply]
sounds good ! thanks I'll try that :)
